src/ontology/phenio.Makefile

```diff
@@ -3,4 +3,28 @@
 ## If you need to customize your Makefile, make
 ## changes here rather than in the main Makefile
 
+# Replace the assembly of phenio-full.owl with a
+# similar process including subq reconstruction
 
+# Constants
+SUBQ_QUERY_PATH=                $(SPARQLDIR)/subq_construct.sparql
+UPDATE_QUERY_PATH=              ./subq_update.sparql
+
+# Full file assembly
+$(ONT)-full.owl: $(SRC) $(OTHER_SRC) $(IMPORT_FILES)
+	$(ROBOT) merge --input $< \
+		reason --reasoner ELK --equivalent-classes-allowed asserted-only --exclude-tautologies structural \
+		relax \
+		reduce -r ELK \
+		$(SHARED_ROBOT_COMMANDS) annotate --ontology-iri $(ONTBASE)/$@ $(ANNOTATE_ONTOLOGY_VERSION) --output $@.tmp.owl && mv $@.tmp.owl $@
+
+	echo "Finding subq patterns based on $(SUBQ_QUERY_PATH)..."
+	$(ROBOT) query --input $@ --format 'owl' --query $(SUBQ_QUERY_PATH) $@_subqs_queryresult.tmp.owl
```

Comment:
Its good practice to store tmp files in the $(TMPDIR) so that when make commands fail, the intermediately created files do not clutter the workspace. Also, Make goal should be minimal and responsible for the creation of one file only. So in your case, you would create a bunch of intermediate make goals (warning this is just freestyle so you get the picture - probably some typing errors):

```
$(TMPDIR)/$(ONT)-full.owl:  $(SRC) $(OTHER_SRC) $(IMPORT_FILES)
	$(ROBOT) merge --input $< \
		reason --reasoner ELK --equivalent-classes-allowed asserted-only --exclude-tautologies structural \
		relax \
		reduce -r ELK \
		$(SHARED_ROBOT_COMMANDS) --output $@.tmp.owl && mv $@.tmp.owl $@

$(TMPDIR)/$(ONT)-full-subq.owl: $(TMPDIR)/$(ONT)-full.owl
	$(ROBOT) query --input $< --format 'owl' --query $(SUBQ_QUERY_PATH) $@

$(UPDATE_QUERY_PATH): $(TMPDIR)/$(ONT)-full-subq.owl
	#echo "Creating update query..."
	awk -v RS= 'NR==1' $(SUBQ_QUERY_PATH) > $@
	echo -e "INSERT DATA\n{\n" >> $@
	tail -n +3 $< >> $@

$(ONT)-full.owl: $(TMPDIR)/$(ONT)-full.owl $(UPDATE_QUERY_PATH)
	$(ROBOT) query --input $< --format 'owl' --update $(UPDATE_QUERY_PATH) --temporary-file 'true' annotate --ontology-iri $(ONTBASE)/$@ $(ANNOTATE_ONTOLOGY_VERSION) 
```

Reply:
✔️